modules/kn-service-offline-create.adoc

```diff
@@ -0,0 +1,108 @@
+* You have installed `kn` CLI.
+
+.Procedure
+
+. Create an offline Knative service:
++
+[source,terminal]
+----
+$ kn service create event-display \
+    --image quay.io/openshift-knative/knative-eventing-sources-event-display:latest \
+    --target ./ \
+    --namespace test
+----
++
+[source,terminal]
+----
+Service 'event-display' created in namespace 'test'.
+----
++
+* The `--namespace test` option places the new service in the `test` namespace.
++
+If you do not use `--namespace`, the service is created in the `default` namespace.
++
+* The `--target ./` flag enables the offline mode and specifies `./` as the directory for storing the new directory tree.
++
+If you do not specify an existing directory, but use a filename, such as `--target my-service.yaml`, then no directory tree is created. Instead, only the service descriptor file `my-service.yaml` is created in the current directory.
++
+Additionally, specifying a filename ending in `.json` results into creating the service descriptor file in the JSON format.
+
+. Examine the created directory structure:
++
+[source,terminal]
+----
+$ tree ./
+./
+└── test
+    └── ksvc
+        └── event-display.yaml
+
+2 directories, 1 file
+----
++
+* The old `./` directory specified with `--target` contains the new `test/` directory, named after the specified namespace.
+* The `test/` directory contains the `ksvc` directory, named after the resource type.
+* The `ksvc` directory contains the file `event-display.yaml`, named according to the specified service name.
+
+. Examine the generated service descriptor file:
++
+[source,terminal]
+----
+$ cat test/ksvc/event-display.yaml
+----
++
+[source,terminal]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  creationTimestamp: null
+  name: event-display
+  namespace: test
+spec:
+  template:
+    metadata:
+      annotations:
+        client.knative.dev/user-image: quay.io/openshift-knative/knative-eventing-sources-event-display:latest
+      creationTimestamp: null
+    spec:
+      containers:
+      - image: quay.io/openshift-knative/knative-eventing-sources-event-display:latest
+        name: ""
+        resources: {}
+status: {}
+----
+
+. List information about the new service:
++
+[source,terminal]
+----
+$ kn service describe event-display --target ./ --namespace test
+Name:       event-display
+Namespace:  test
+Age:
+URL:
+
+Revisions:
+
+Conditions:
+  OK TYPE    AGE REASON
+----
++
+* The `--namespace` option specifies the namespace, which communicates to `kn` the subdirectory that contains the necessary service descriptor file.
++
+Omitting the `--namespace` option makes `kn` search for the service in the `default/` subdirectory.
++
+* The `--target ./` option specifies the root directory for the directory structure containing namespace subdirectories.
++
+Alternatively, you can directly specify a YAML or JSON filename with the `--target` option.
```

Comment:
I'd add a list of accepted file extensions that are required to recognize the file mode operation. And could we highlight it a bit more by moving to step no. 1 description maybe? The file mode should be a preferred way for quick single service generation etc. 

E.g.:
```suggestion
Alternatively, you can directly specify a YAML or JSON filename with the `--target` option. The accepted file extensions are  `.yaml, .yml, .json`.
```

Reply:
+1

Reply:
@dsimansk I've added the list of available extensions to the first step. Notice that we already are talking about the "single file" option in the first step. But I've now made it the first option we explain (before `--namespace`).